spec/unit/messages/deployment_create_message_spec.rb

```diff
@@ -127,6 +127,160 @@ module VCAP::CloudController
+          body['options'] = { canary: {} }
+          body['strategy'] = 'rolling'
+          message = DeploymentCreateMessage.new(body)
+          expect(message).not_to be_valid
+          expect(message.errors[:'options.canary']).to include('are only valid for Canary deployments')
+        end
+
+        it 'errors when there is an unknown option' do
+          body['options'] = { foo: 'bar', baz: 'boo' }
+          message = DeploymentCreateMessage.new(body)
+          expect(message).not_to be_valid
+          expect(message.errors[:options]).to include('has unsupported key(s): foo, baz')
+        end
+
+        context 'steps' do
+          it 'errors when is not an array' do
+            body['options'] = { canary: { steps: 'foo' } }
+            message = DeploymentCreateMessage.new(body)
+            expect(message).not_to be_valid
+            expect(message.errors[:'options.canary.steps']).to include('must be an array of objects')
+          end
+
+          it 'is valid when is an empty array' do
+            body['options'] = { canary: { steps: [] } }
+            message = DeploymentCreateMessage.new(body)
+            expect(message).to be_valid
+          end
+
+          it 'is valid when is nil' do
+            body['options'] = { canary: { steps: nil } }
+            message = DeploymentCreateMessage.new(body)
+            expect(message).to be_valid
+          end
+
+          it 'errors if not an array of objects' do
+            body['options'] = { canary: { steps: [{ instance_weight: 1 }, 'foo'] } }
+            message = DeploymentCreateMessage.new(body)
+            expect(message).not_to be_valid
+            expect(message.errors[:'options.canary.steps']).to include('must be an array of objects')
+          end
+
+          it 'errors if steps have an unsupported key' do
+            body['options'] = { canary: { steps: [{ instance_weight: 1 }, { instance_weight: 1, foo: 'bar', baz: 1 }, { baz: 1 }] } }
+            message = DeploymentCreateMessage.new(body)
+            expect(message).not_to be_valid
+            expect(message.errors[:'options.canary.steps']).to include('has unsupported key(s): foo, baz')
+            expect(message.errors[:'options.canary.steps']).to include('has unsupported key(s): baz')
+          end
+
+          context 'instance_weights' do
+            it 'is valid if instance_weights are Integers between 1-100 in ascending order' do
+              body['options'] = { canary: { steps: [{ instance_weight: 1 }, { instance_weight: 2 }, { instance_weight: 50 }, { instance_weight: 99 }, { instance_weight: 100 }] } }
+              message = DeploymentCreateMessage.new(body)
+              expect(message).to be_valid
+            end
+
+            it 'is valid if there are duplicate instance_weights' do
+              body['options'] = { canary: { steps: [{ instance_weight: 10 }, { instance_weight: 10 }, { instance_weight: 50 }, { instance_weight: 50 }] } }
```

Comment:
was there a reason to support this?

Reply:
As opposed to just rejecting the request? We can do that but I don't think its essential.